types.ts

```diff
@@ -413,6 +418,30 @@ export interface WebBasicAuthentication extends BaseAuthentication {
   };
 }
 
+export interface CustomAuthParameter {
+  /**
+   * The name used to refer to the parameter. Should match the string for the key in the params mapping.
+   */
+  name: string;
+
+  /**
+   * A description shown to the user indicating what value they should provide for this parameter.
+   */
+  description: string;
+}
+
+/**
+ * Authenticate using custom templated parameters, which you insert into the request. The user provides all the
```

Comment:
Maybe something like:

Authenticate in a custom way by having one or more arbitrary secret values inserted into the request url or body using template replacement.

Some APIs use non-standard authentication schemes which often require secret credentials to be put in specific places in the request URL or request body. Custom authentication supports many of these cases by allowing you as the pack author to define one or more secret values that the user must provide (or that you as the pack author must provide, if using system authentication), and then you can specify where these values are to be inserted into the request using a syntax using curly braces inspired by template engines.

The syntax to use for inserting a secret credential is `{{<paramName>-<invocationToken>}}`. That is, curly braces surrounding the name of one of the parameters you define, with a suffix that is the unique invocation token for the current formula invocation. The invocation token is part of the {@link ExecutionContext} and is required for security reasons.

@example
```
// Suppose you're using an API that wants a secret id in the request URL,
// and a secret value in the request body. You can define a Custom authentication
// configuration with two params, one called "secretId" and one called "secretValue".
// The user will be prompted to specify a value for each of these when setting up an account.
// In the `execute` body of your formula, you can specify where those values are
// inserted into the request using curly brace {{ }} template replacement syntax.
// The string that you put inside the curly braces is the name of a parameter
// (in this case either "secretId" or "secretValue"), then a dash, then the value
// of `context.invocationToken`. This makes the actual template replacement string
// unique to this formula invocation, which is important for security reasons.
execute: async function([], context) {
  let secretIdTemplateName = "secretId-" + context.invocationToken;
  let urlWithSecret = "/api/entities/{{" + secretIdTemplateName + "}}"

  let secretValueTemplateName = "secretValue-" + context.invocationToken;
  let bodyWithSecret = JSON.stringify({
    key: "{{" + secretValueTemplateName + "}}",
    otherBodyParam: "foo",
  });
 
  let response = await context.fetcher.fetch({method: "GET", url: urlWithSecret, body: bodyWithSecret});
  ...
}
```

Reply:
oh this is amazing thanks for the suggestion! The example looks great -- any thoughts on whether we should include a link to real API examples of when you might need this? (i.e. link to plaid and briefly 1-2 sentence describe it requires secret in both the url and body)